packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signature: string;
+        }
+    }>;
+    signAllTransactions(messages: string[]): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signatures: string[];
+        }
+    }>;
+}
+
+interface SlopeWindow extends Window {
+    Slope?: any;
```

Comment:
This doesn't seem like it should be typed as `any`? The interface is provided.

Reply:
I'm not sure how to modify this, changing `any` to another value will cause `new window.Slope()` to report an error.

Reply:
Since `Slope` has a constructor function, this might do what you want.
```suggestion
    Slope?: {
        new (): SlopeWallet;
    };
```